golang/trace/zipkin_tracereporter.go

```diff
@@ -0,0 +1,100 @@
+package trace
+
+import (
+	"encoding/base64"
+	"encoding/binary"
+	"net"
+	"time"
+
+	tc "github.com/uber/tchannel/golang"
+	"github.com/uber/tchannel/golang/thrift"
+	"github.com/uber/tchannel/golang/trace/thrift/gen-go/tcollector"
+)
+
+// ZipkinTraceReporter is a trace reporter that submits trace spans in to zipkin trace server.
+type ZipkinTraceReporter struct {
+	tchannel *tc.Channel
+}
+
+// Report method will submit trace span to tcollector server.
+func (r *ZipkinTraceReporter) Report(
```

Comment:
Should spans be reported synchronously? Do we need to block our tchannel call till tcollector gets the span?

(We can change this in a separate change, I just want to make sure we're thinking about how this could be done in the background)